### PR TITLE
Chore/add events to configmap

### PR DESCRIPTION
**Pull request by j1m-ryan**
### Proposed changes

Add events to main Nginx Config ConfigMap

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

**Review by jjngx — APPROVED** [review]

File: cmd/nginx-ingress/main.go [hunk 1]
```diff
@@ -865,7 +865,7 @@ func processConfigMaps(kubeClient *kubernetes.Clientset, cfgParams *configs.Conf
 		if err != nil {
 			nl.Fatalf(l, "Error when getting %v: %v", *nginxConfigMaps, err)
 		}
-		cfgParams = configs.ParseConfigMap(cfgParams.Context, cfm, *nginxPlus, *appProtect, *appProtectDos, *enableTLSPassthrough)
+		cfgParams, _ = configs.ParseConfigMap(cfgParams.Context, cfm, *nginxPlus, *appProtect, *appProtectDos, *enableTLSPassthrough, eventLog)
```
Comment (pdabelf5) [inline cmd/nginx-ingress/main.go:868]: should we handle this `ok` bool?

File: internal/configs/configmaps.go [hunk 2]
```diff
@@ -416,6 +490,7 @@ func ParseConfigMap(ctx context.Context, cfgm *v1.ConfigMap, nginxPlus bool, has
 	if keepaliveRequests, exists, err := GetMapKeyAsInt64(cfgm.Data, "keepalive-requests", cfgm); exists {
 		if err != nil {
 			nl.Error(l, err)
+			configOk = false
```
Comment (pdabelf5) [inline internal/configs/configmaps.go:493]: why no event for this one?

File: cmd/nginx-ingress/main.go (hunk 1 above)
Comment (j1m-ryan) [reply cmd/nginx-ingress/main.go:868]: if we make an event here, there will be two events the first time i check the configmap

File: internal/configs/configmaps.go (hunk 2 above)
Comment (j1m-ryan) [reply internal/configs/configmaps.go:494]: Missed this, Thanks, added now.

**Review by pdabelf5 — APPROVED** [review]

**Review by AlexFenlon — APPROVED** [review]

**Pull request merged**